x-pack/plugins/upgrade_assistant/public/application/components/overview/upgrade_system_indices/upgrade_system_indices.tsx

```diff
@@ -0,0 +1,205 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FunctionComponent, useEffect } from 'react';
+
+import { i18n } from '@kbn/i18n';
+import type { DocLinksStart } from 'src/core/public';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiText,
+  EuiButton,
+  EuiSpacer,
+  EuiLink,
+  EuiIcon,
+  EuiButtonEmpty,
+  EuiCallOut,
+  EuiFlexGroup,
+  EuiFlexItem,
+} from '@elastic/eui';
+import type { EuiStepProps } from '@elastic/eui/src/components/steps/step';
+
+import type { OverviewStepProps } from '../../types';
+import { useSystemIndicesUpgrade } from './use_system_indices_upgrade';
+
+interface Props {
+  setIsComplete: OverviewStepProps['setIsComplete'];
+}
+
+interface StepProps extends OverviewStepProps {
+  docLinks: DocLinksStart;
+}
+
+const i18nTexts = {
+  title: i18n.translate('xpack.upgradeAssistant.overview.system_indices.title', {
+    defaultMessage: 'Upgrade system indices',
+  }),
+  bodyDescription: (docLink: string) => (
+    <FormattedMessage
+      id="xpack.upgradeAssistant.overview.system_indices.body"
+      defaultMessage="Upgrade your system indices to keep them happy. In addition to regular checkups, it is recommended that you brush and floss your indices twice per day. {learnMoreLink}."
+      values={{
+        learnMoreLink: (
+          <EuiLink href={docLink} target="_blank">
+            Learn more
+          </EuiLink>
+        ),
+      }}
+    />
+  ),
+  startButtonLabel: i18n.translate(
+    'xpack.upgradeAssistant.overview.system_indices.startButtonLabel',
+    {
+      defaultMessage: 'Begin upgrading system indices',
+    }
+  ),
+  inProgressButtonLabel: i18n.translate(
+    'xpack.upgradeAssistant.overview.system_indices.inProgressButtonLabel',
+    {
+      defaultMessage: 'Upgrading system indices',
+    }
+  ),
+  noUpgradeNeeded: i18n.translate(
+    'xpack.upgradeAssistant.overview.system_indices.noUpgradeNeeded',
+    {
+      defaultMessage: 'All system indices have been upgraded',
+    }
+  ),
+  viewSystemIndicesStatus: i18n.translate(
+    'xpack.upgradeAssistant.system_indices.overview.system_indices.viewSystemIndicesStatus',
+    {
+      defaultMessage: 'View status',
+    }
+  ),
+  retryButtonLabel: i18n.translate(
+    'xpack.upgradeAssistant.overview.system_indices.retryButtonLabel',
+    {
+      defaultMessage: 'Try again',
+    }
+  ),
+  loadingError: i18n.translate('xpack.upgradeAssistant.overview.system_indices.loadingError', {
+    defaultMessage: 'An error occurred while retrieving the status of system indices',
+  }),
+};
+
+const UpgradeSystemIndicesStep: FunctionComponent<Props> = ({ setIsComplete }) => {
+  const { beginSystemIndicesUpgrade, startUpgradeStatus, upgradeStatus, setShowFlyout } =
+    useSystemIndicesUpgrade();
+
+  useEffect(() => {
+    if (upgradeStatus.data?.upgrade_status === 'NO_UPGRADE_NEEDED') {
+      setIsComplete(true);
```

Comment:
Would we ever want to set this back to `false`? Say, if `upgradeStatus.data?.upgrade_status` changes to another value?

Reply:
nice catch!